wallet-sample-app-practice/main/Asset/daml/Asset.daml

```diff
@@ -0,0 +1,47 @@
+module Asset where 
+
+import DA.Set as Set
+
+data AssetType = AssetType with 
+    issuer: Party 
+    symbol: Text 
+    fungible: Bool -- true means able to split and merge
+    reference: Optional Text 
+  deriving (Eq, Show) 
+
+template Asset with 
+    assetType: AssetType 
+    owner: Party 
+    amount: Decimal 
+    observers: Set Party
+  where 
+    signatory assetType.issuer, owner 
+    observer observers 
+    ensure (if assetType.fungible then amount > 0.0 else amount == 1.0)
+    -- determined by amount so can't use key?
+
+template AssetTransfer with 
+    asset: Asset 
```

Comment:
why do we want to pass the whole Asset here instead of just cid?

Reply:
when we compare assets we don't need to compare observers, so if we use assetCid and the observers change it will lead to stale cid. (although currently there is no choice to update observers)